# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.cpp
# Commit: 88654fa3af79
# Full Hash: 88654fa3af790e408f6b3ec6b9aa5b203937b36e
# Author: stransky <stransky@redhat.com>
# Date: 2021-01-31 09:52:27
# Description:
#   Bug 1689623 [Linux] Check mWidget before use at GtkCompositorWidget::GetTransparentRegion(), r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D103554
# ==============================================================================

diff -r d9e5a018312e -r 88654fa3af79 widget/gtk/GtkCompositorWidget.cpp
--- a/widget/gtk/GtkCompositorWidget.cpp	Sat Jan 30 15:55:00 2021 +0000
+++ b/widget/gtk/GtkCompositorWidget.cpp	Sat Jan 30 15:55:18 2021 +0000
@@ -126,6 +126,10 @@
 #endif
 
 LayoutDeviceIntRegion GtkCompositorWidget::GetTransparentRegion() {
+  if (!mWidget) {
+    return LayoutDeviceIntRect();
+  }
+
   // We need to clear target buffer alpha values of popup windows as
   // SW-WR paints with alpha blending (see Bug 1674473).
   if (mWidget->IsPopup()) {
